### PR TITLE
tests: Add fuzzing harnesses for functions in addrdb.h, net_permissions.h and timedata.h

### PR DESCRIPTION
**Pull request by practicalswift**
Add fuzzing harnesses for functions in `addrdb.h`, `net_permissions.h` and `timedata.h`.

### CODE REVIEW

File: src/test/fuzz/timedata.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
+    const unsigned int max_size = fuzzed_data_provider.ConsumeIntegralInRange<unsigned int>(0, 1000);
+    // Divide by 2 to avoid signed integer overflow in .median()
+    const int64_t initial_value = fuzzed_data_provider.ConsumeIntegral<int64_t>() / 2;
+    CMedianFilter<int64_t> median_filter{max_size, initial_value};
+    while (fuzzed_data_provider.remaining_bytes() > 0) {
+        (void)median_filter.median();
+        assert(median_filter.size() > 0);
+        assert(static_cast<size_t>(median_filter.size()) == median_filter.sorted().size());
+        assert(static_cast<unsigned int>(median_filter.size()) <= max_size || max_size == 0);
```
Comment (MarcoFalke) [inline src/test/fuzz/timedata.cpp]: Why is this needed?

```suggestion
        assert(static_cast<unsigned int>(median_filter.size()) <= max_size);
```

File: src/test/fuzz/net_permissions.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#include <cassert>
+#include <cstdint>
+#include <string>
+#include <vector>
+
+void test_one_input(const std::vector<uint8_t>& buffer)
+{
+    FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
+    const std::string s = fuzzed_data_provider.ConsumeRandomLengthString(32);
+    const NetPermissionFlags net_permission_flags = fuzzed_data_provider.ConsumeBool() ? fuzzed_data_provider.PickValueInArray<NetPermissionFlags>({NetPermissionFlags::PF_NONE, NetPermissionFlags::PF_BLOOMFILTER, NetPermissionFlags::PF_RELAY, NetPermissionFlags::PF_FORCERELAY, NetPermissionFlags::PF_NOBAN, NetPermissionFlags::PF_MEMPOOL, NetPermissionFlags::PF_ISIMPLICIT, NetPermissionFlags::PF_ALL}) : static_cast<NetPermissionFlags>(fuzzed_data_provider.ConsumeIntegral<uint32_t>());
```
[the remark this answers is not in the document]
Comment (practicalswift) [reply src/test/fuzz/net_permissions.cpp]: Now fixed. Considering updating `.clang-format` regarding this whitespace nit since it seems to be recurring (I agree with your rationale).

Ideally the `.clang-format` should be written such that `clang-format -i` makes a PR immune against whitespace nits :)

File: src/test/fuzz/timedata.cpp (hunk 1 above)
Comment (practicalswift) [reply src/test/fuzz/timedata.cpp]: It is needed since `max_size == 0` means unlimited size :)

File: src/test/fuzz/addrdb.cpp [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            const Optional<CBanEntry> ban_entry = ConsumeDeserializable<CBanEntry>(fuzzed_data_provider);
+            if (ban_entry) {
+                return *ban_entry;
+            }
+            break;
+        }
+        }
+        return CBanEntry{};
+    }();
+    assert(!ban_entry.banReasonToString().empty());
```
Comment (MarcoFalke) [inline src/test/fuzz/addrdb.cpp]: Would it make sense to also add an `AssertEqualAfterSerializeDeserialize` here?

File: src/test/fuzz/net_permissions.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (MarcoFalke) [reply src/test/fuzz/net_permissions.cpp]: I think the goal of clang-format is to only add whitespace, not commas or even normalize whitespace like black does for python

File: src/test/fuzz/addrdb.cpp (hunk 3 above)
Comment (practicalswift) [reply src/test/fuzz/addrdb.cpp]: I'm afraid equality is not defined for `CBanEntry`. That goes for quite a few of our classes :)
